File: src/bin/01-1.hs
main :: IO ()
main = do
  contents <- readFile "input.txt"
  let result = process contents
  putStrLn result

process :: String -> String
process input = show (countIncs measurements)
  where
    measurements = map read $ lines input

countIncs :: (Num a, Ord a) => [a] -> Int
countIncs xs = length $ filter (> 0) deltas
  where
    deltas = map (\(a, b) -> b - a) pairs
    pairs = zip xs (drop 1 xs)

File: src/bin/01-1.rs
use std::io::BufRead;
use std::ops::Sub;

fn main() {
    let input = std::io::stdin()
        .lock()
        .lines()
        .map(|line| line.unwrap().parse().unwrap())
        .collect();
    println!("{}", count_incs(&input));
}

fn count_incs(input: &Vec<i32>) -> usize {
    let pairs = input.iter().zip(input.iter().skip(1));
    let deltas = pairs.map(|(a, b)| b.sub(a));
    deltas.filter(|x| x.is_positive()).count()
}
